upgrade template-repo to slidev-v52.0.0, node 24, pnpm 10.13.1

fernan opened the pull request:
Hi @kxxt ,

first of all thanks for a great template! That was exactly what I needed, and what all the sli.dev 'getting started' tutorials were missing. 

I will keep adjusting to my needs (collections of presentations for teaching), but for now here are the changes I needed to make this run without errors with more recent versions of slidev, node, pnpm. 

 * pnpm-lock.yaml
   - here are the largest changes, I just literally copied over the pnpm-lock file from a fresh sli.dev installation with pnpm
 * pnpm-workspace.yml
   - when installing with `pnpm install` I got a warning about the necessity to `pnpm approve-builds` and esbuild was listed there as an option, when done I say this change added to pnpm-worskpace. Maybe this should have to be added to the README.md for completeness?
 * README.md
   - fix the pnpm flag, I see pnpm currently uses -C instead of -F ([source](https://pnpm.io/pnpm-cli))
 * slides.md, imported-slides.md, snippets/external.ts
   - these are all coming from the newer sli.dev demo slides
 * private package.json for presentations
   - update versions to match 
 * deploy.yml
   - here I just updated actions to the most recent versions, not sure about the github-pages-directory-listing, I must confess I'm a complete newbie on this, I was just googling for 'github-pages-directory-listing' and came with jayanta525's version, not sure if this at v4 is better in any way than your own at v2.

Hope this helps and thanks again for putting this together! Best -- fernan

fernan committed:
upgrade template-repo to slidev-v52.0.0, node 24, pnpm 10.13

kxxt commented:
Thanks for the PR!

I am a little busy recently so I will only review part of it for now and come back later.

> here I just updated actions to the most recent versions, not sure about the github-pages-directory-listing, I must confess I'm a complete newbie on this, I was just googling for 'github-pages-directory-listing' and came with jayanta525's version, not sure if this at v4 is better in any way than your own at v2.

Please keep the fork. My version contains fixes necessary for preventing apindex from overwriting slidev's generated `index.html` with a directory listing.

fernan committed:
revert change of github-pages-directory-listing

We go back to kxxt's v2, as suggested by kxxt in the comment to the PR.

fernan commented:
Thanks for the fast reponse. I've just committed a reversion to your own fork of github-pages-directory-listing. I think that completes this PR. Just wanted to help get this updated. Whenever you have time ... Thanks again!

kxxt commented on README.md:
```diff
@@ -18,7 +18,7 @@ Example: [Slides repo for my course projects](https://lie-flat.github.io/slides/
 4. Enable the `deploy` workflow in GitHub Actions
 5. Duplicate the `template` folder in the `slides` folder and rename it to your presentation name. (Don't forget to update the id in `package.json`)
-6. Edit the slides and optionally preview them locally with `pnpm -F <slide-subfolder-name> run dev` 
+6. Edit the slides and optionally preview them locally with `pnpm -C <slide-subfolder-name> run dev` 
```
> fix the pnpm flag, I see pnpm currently uses -C instead of -F ([source](https://pnpm.io/pnpm-cli))

I don't think that's correct. `-F` filters the package while `-C` changes the current directory.

So both `pnpm -F <slide-subfolder-name> run dev` and `pnpm -C <PATH/TO/slide-folder> run dev` works but
`pnpm -C <slide-subfolder-name> run dev` won't work unless you place your slide folders at top level of the git repo.

fernan commented:
Oh I see. Thanks for the detailed response. I had trouble finding the
documentation on -F vs -C, as you can guess from this issue (
https://github.com/pnpm/pnpm/issues/9756).

I guess -F is described here? https://pnpm.io/filtering. That makes sense
then.

On Sat, Jul 19, 2025 at 10:55 AM Levi Zim ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/kxxt/slidev-template-repo/pull/2#discussion_r2217324429>
> :
>
> > @@ -18,7 +18,7 @@ Example: [Slides repo for my course projects](https://lie-flat.github.io/slides/
>  3. Change "workflow permissions" to "Read and write permission" in `Repo Settings` > `Actions` > `General`.
>  4. Enable the `deploy` workflow in GitHub Actions
>  5. Duplicate the `template` folder in the `slides` folder and rename it to your presentation name. (Don't forget to update the id in `package.json`)
> -6. Edit the slides and optionally preview them locally with `pnpm -F <slide-subfolder-name> run dev`
> +6. Edit the slides and optionally preview them locally with `pnpm -C <slide-subfolder-name> run dev`
>
> fix the pnpm flag, I see pnpm currently uses -C instead of -F (source
> <https://pnpm.io/pnpm-cli>)
>
> I don't think that's correct. -F filters the package while -C changes the
> current directory.
>
> So both pnpm -F <slide-subfolder-name> run dev and pnpm -C
> <PATH/TO/slide-folder> run dev works but
> pnpm -C <slide-subfolder-name> run dev won't work unless you place your
> slide folders at top level of the git repo.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kxxt/slidev-template-repo/pull/2#pullrequestreview-3035378271>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABK247AF5ZVFWBMYRJK3WL3JJE47AVCNFSM6AAAAACBSIBW7OVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTAMZVGM3TQMRXGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
--
fernan

// If not controlled, work will flow to the competent ones
    until they submerge -- Charles Boyle //

fernan committed:
revert the change on -F vs -C

my bad, did not understood the use of -F --filtering [pnpm
filtering](https://pnpm.io/filtering) correctly.

kxxt committed:
Increase heap size for nodejs to avoid OOM when building slides

kxxt reviewed (approved):
LGTM, I finally got some time to properly review and test this PR. Thanks for your contribution and sorry for the long delay.

When testing this PR, I noticed that the CI build fails due to Node.js heap OOM, so I pushed a commit for increasing the heap size.
